### PR TITLE
Free apps from using `github.com/nspcc-dev/neofs-api-go/v2` module

### PR DESCRIPTION
**Pull request by cthulhu-rider**
https://github.com/nspcc-dev/neo-go/ imports indirectly, all fine. https://github.com/nspcc-dev/neofs-node is maintained in a bundle and can be cleaned up last (not in this PR)

projects:
* https://github.com/nspcc-dev/neofs-rest-gw/
* https://github.com/nspcc-dev/neofs-s3-gw

### CODE REVIEW

File: netmap/filter.go [hunk 1]
```diff
@@ -13,17 +11,18 @@ const mainFilterName = "*"
 
 // processFilters processes filters and returns error is any of them is invalid.
 func (c *context) processFilters(p PlacementPolicy) error {
-	for i := range p.filters {
-		if err := c.processFilter(p.filters[i], true); err != nil {
-			return fmt.Errorf("process filter #%d (%s): %w", i, p.filters[i].GetName(), err)
+	fs := p.Filters()
```
Comment (roman-khimov) [inline netmap/filter.go:14]: This makes a copy, but I doubt we need it.

File: netmap/netmap.go [hunk 2]
```diff
@@ -178,7 +178,7 @@ func (m NetMap) PlacementVectors(vectors [][]NodeInfo, objectID oid.ID) ([][]Nod
 // Make a copy if you need to change it.
 func (m NetMap) ContainerNodes(p PlacementPolicy, containerID cid.ID) ([][]NodeInfo, error) {
 	c := newContext(m)
-	c.setCBF(p.backupFactor)
+	c.setCBF(p.ContainerBackupFactor())
```
Comment (roman-khimov) [inline netmap/netmap.go:181]: Not required.

File: netmap/netmap.go [hunk 3]
```diff
@@ -192,15 +192,16 @@ func (m NetMap) ContainerNodes(p PlacementPolicy, containerID cid.ID) ([][]NodeI
 		return nil, err
 	}
 
-	result := make([][]NodeInfo, len(p.replicas))
-
-	for i := range p.replicas {
-		sName := p.replicas[i].GetSelector()
+	rs := p.Replicas()
```
Comment (roman-khimov) [inline netmap/netmap.go:195]: This also makes a copy that we don't need.

File: netmap/netmap.go [hunk 4]
```diff
@@ -192,15 +192,16 @@ func (m NetMap) ContainerNodes(p PlacementPolicy, containerID cid.ID) ([][]NodeI
...
+	result := make([][]NodeInfo, len(rs))
+	for i := range rs {
+		sName := rs[i].SelectorName()
 		if sName == "" {
-			if len(p.selectors) == 0 {
-				var s netmap.Selector
-				s.SetCount(p.replicas[i].GetCount())
-				s.SetFilter(mainFilterName)
+			ss := p.Selectors()
```
Comment (roman-khimov) [inline netmap/netmap.go:200]: Here as well.

File: netmap/selector.go [hunk 5]
```diff
@@ -5,25 +5,25 @@ import (
...
 
 	"github.com/nspcc-dev/hrw/v2"
-	"github.com/nspcc-dev/neofs-api-go/v2/netmap"
 )
 
 // processSelectors processes selectors and returns error is any of them is invalid.
 func (c *context) processSelectors(p PlacementPolicy) error {
-	for i := range p.selectors {
-		fName := p.selectors[i].GetFilter()
+	ss := p.Selectors()
```
Comment (roman-khimov) [inline netmap/selector.go:12]: Copy.

File: netmap/filter.go (hunk 1 above)
Comment (cthulhu-rider) [reply netmap/filter.go]: reverted all

File: netmap/policy.go [hunk 6]
```diff
@@ -234,23 +234,21 @@ func (r ReplicaDescriptor) SelectorName() string {
 	return r.m.GetSelector()
 }
 
-// AddReplicas adds a bunch object replica's characteristics.
+// SetReplicas sets list of object replica's characteristics.
```
Comment (roman-khimov) [inline netmap/policy.go:237]: Not sure it's an improvement, but let's see how it goes.

**Review by roman-khimov — APPROVED** [review]

**Review by smallhive — APPROVED** [review]

File: eacl/filter.go [hunk 7]
```diff
@@ -13,35 +13,25 @@ import (
...
-	fKeyObjID
-	fKeyObjContainerID
-	fKeyObjOwnerID
-	fKeyObjCreationEpoch
-	fKeyObjPayloadLength
-	fKeyObjPayloadHash
-	fKeyObjType
-	fKeyObjHomomorphicHash
-	fKeyObjLast // helper, used in tests
+	FilterObjectVersion                    = v2acl.FilterObjectVersion
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply eacl/filter.go:26]: naming as much straightforward, i dont think writing `FilterObjectVersion // object version` is needed

**Pull request merged**